### PR TITLE
Added support for scikit learn gradient boosted trees

### PR DESCRIPTION
**Pull request by jverre-drivalytix**
Changes:
* Added support for scikit learn gradient boosted trees (#232 )
* Fixed bug around undefined variable names

### CODE REVIEW

File: shap/explainers/tree.py [hunk 1]
```diff
@@ -292,6 +304,7 @@ def shap_interaction_values(self, X, tree_limit=-1, **kwargs):
 
     def _tree_shap_ind(self, i):
         phi = np.zeros((self._current_X.shape[1] + 1, self.n_outputs))
+        print(self.base_offset)
```
Comment (slundberg) [inline shap/explainers/tree.py]: should remove this.

File: shap/explainers/tree.py [hunk 2]
```diff
@@ -75,7 +75,19 @@ def __init__(self, model, **kwargs):
...
+
+            scale = len(model.estimators_) * model.learning_rate
+            self.trees = [Tree(e.tree_, scaling=scale) for e in model.estimators_[:,0]]
+            self.less_than_or_equal = True
+        elif str(type(model)).endswith("sklearn.ensemble.gradient_boosting.GradientBoostingClassifier'>"):
+            
+            # currently we only support the logs odds estimator
+            if str(type(model.init_)).endswith("ensemble.gradient_boosting.LogOddsEstimator'>"):
+                self.base_offset = model.init_.prior
+                print('base_offset =', self.base_offset)
```
Comment (slundberg) [inline shap/explainers/tree.py]: should remove the print statement

**Comment by slundberg** [comment]
Thanks! Could you also add a unit test in `test_tree.py` just to make sure this works?

**Comment by jverre-drivalytix** [comment]
I have removed the print statements.

I have also added two unit tests to test_tree.py that are inline with the RandomForestClassifier unit tests

**Comment by slundberg** [comment]
Thanks again, looks great!
